src/rez/pip.py

```diff
@@ -81,35 +86,106 @@ def find_pip(pip_version=None, python_version=None):
     """
     py_exe = None
     context = None
+    found_pip_version = None
 
-    py_exe, pip_version, context = find_pip_from_context(
-        python_version,
-        pip_version=pip_version
-    )
-
-    if not py_exe:
-        py_exe, pip_version, context = find_pip_from_context(
-            python_version,
-            pip_version=pip_version or "latest"
-        )
+    for version in [pip_version, pip_version or "latest"]:
+        try:
+            py_exe, found_pip_version, context = find_pip_from_context(
+                python_version, pip_version=version
+            )
+            if _check_found(py_exe, found_pip_version):
+                break
+        except BuildError as error:
+            print_warning(str(error))
 
-    if not py_exe:
+    else:  # No break from: for version in [pip_version,...]
```

Comment:
I can't believe I've been writing python this long and have never seen this. There's nothing technically wrong with this, but tbh I'd avoid it because it's so esoteric.

Reply:
True, I also find it fairly susceptible to indent/whitespace messing up logic if code was reformatted/refactored incorrectly hence the comment. I'll see if I can change it up a bit